prog/dftb+/lib_math/ranlux.F90

```diff
@@ -268,6 +269,23 @@ subroutine Ranlux_getRandom2DArray(self, r2Darray)
 
   end subroutine Ranlux_getRandom2DArray
 
+  !> Fills a given 3D array with random numbers.
+  subroutine Ranlux_getRandom3DArray(self, r3Darray)
+
+    !> Ranlux instance
+    type(ORanlux), intent(inout) :: self
+
+    !> Array containing the random numbers on exit.
+    real(dp), intent(out) :: r3Darray(:,:,:)
+
+    real(dp), allocatable :: rvec(:)
+
+    allocate(rvec(product(shape(r3Darray))))
+    call getRandomVector_local(rvec, self%iseeds, self%icarry, self%in24, &
+        &self%i24, self%j24, self%next, self%nskip, self%twom24, self%twom12)
+    r3Darray = reshape(rvec,shape(r3Darray))
```

Comment:
`r3dArray(:,:,:) = ...` Although there is no possibility for automatic reallocation on the LHS as `r3dArray` is not allocatable, we should always indicate the array indices on the LHS if no reallocation is desired.